tests/lib/rules/no-computed-in-data.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * @fileoverview Ensure computed properties are not used in the data()
+ * @author IWANABETHATGUY
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+var rule = require('../../../lib/rules/no-computed-in-data')
+
+var RuleTester = require('eslint').RuleTester
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2019,
+    sourceType: 'module'
+  }
+})
+ruleTester.run('no-computed-in-data', rule, {
+
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {
+          return {
+            a: this.b
+          }
+        },
+        computed: {
+          test() {},
+          'foo': {
+
+          }
+        }
+      }
+      </script>
+      `
+    },
+    // should not warn when use computed in methods
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {
+          return {
+            a: this.b
+          }
+        },
+        computed: {
+          test() {},
+        },
+        methods: {
+          foo() {
+            this.test
+          }
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {
+          return {
+            data() {
+              return {
+                test: this.foo
+              }
+            },
+            computed: {
+              foo() {}
+            }
+          }
+        },
+        computed: {
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data() {
+          return {
+            fn() { return this.foo }
+          }
+        },
+        computed: {
+          foo() {}
+        }
+      }
+      </script>
+      `
+    },
+    // should not warn when objectExpression is not a vue component
```

Comment:
@ota-meshi test for https://github.com/vuejs/eslint-plugin-vue/pull/1075/files#r400643684